lib/active_merchant/billing/gateways/banwire.rb

```diff
@@ -81,23 +107,31 @@ def parse(body)
         JSON.parse(body)
       end
 
-      def commit(money, parameters)
-        raw_response = ssl_post(URL, post_data(parameters))
+      def url(action)
+        action == 'purchase' ? URL : URL_STORE
+      end
+
+      def commit(money = nil, parameters, action)
+        raw_response = ssl_post(url(action), post_data(parameters))
         begin
           response = parse(raw_response)
         rescue JSON::ParserError
           response = json_error(raw_response)
         end
 
-        Response.new(success?(response),
+        Response.new(success?(action, response),
                      response["message"],
                      response,
                      :test => test?,
-                     :authorization => response["code_auth"])
+                     :authorization => authorization_from_response(action, response))
+      end
+
+      def authorization_from_response(action, response)
+        action == 'store' ? response['token'] : response['code_auth']
       end
 
-      def success?(response)
-        (response["response"] == "ok")
+      def success?(action, response)
```

Comment:
Hi @bruno, after a first inspection I could determine that this isn't an override of a method in the parent class and it's used only internally, so the parameters order doesn't matter. But after looking the other implementations, the only one that I could find that has a "success?" method with more than 1 parameter is the `orbital` gateway, and it seems that the convention is to pass the **response** as the first parameter, and any other aditional parameters after that.

Maybe I'm just being pedantic, but to keep consistency with the other gateway implementations we could change this to `def success?(response, action)`.

It's up-to-you my friend.